core/base/src/TStorage.cxx

```diff
@@ -200,8 +200,10 @@ void *TStorage::ReAlloc(void *ovp, size_t size)
 #else
    void *vp = ::operator new(size);
 #endif
-   if (vp == nullptr)
-      Fatal(where, "%s", gSpaceErr);
+   if (vp == nullptr) {
+     Fatal(where, "%s", gSpaceErr);
```

Comment:
`Fatal()` shouldn't return, no?  Instead, perhaps it's worth annotating as such (i.e. `[[noreturn]]`) so that compiler and static analysis tools know.
@pcanal what do you think?

Reply:
I confirm that adding __attribute__((noreturn)) on TError.h makes the static analysis warning disappear

Reply:
Side note, from the docu:

```
// Fatal() *might* not abort the program (if gAbortLevel > kFatal) - but for all
// reasonable settings it *will* abort. So let's be reasonable wrt Coverity:
// coverity[+kill]
```

Reply:
In fact, I think that this part of the documentation is wrong or outdated, it should say `gErrorIgnoreLevel` instead of `gAbortLevel`

Reply:
Maybe this part should be removed?

```
// So let's be reasonable wrt Coverity:
// coverity[+kill]
```